internal/manifest/cr_deletion_check.go

```diff
@@ -0,0 +1,51 @@
+package manifest
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	declarative "github.com/kyma-project/lifecycle-manager/internal/declarative/v2"
+	"github.com/kyma-project/lifecycle-manager/pkg/util"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewCustomResourceDeletionCheck creates a check that verifies that the Resource CR in the remote cluster is deleted.
+func NewCustomResourceDeletionCheck() *CRDeletionCheck {
+	return &CRDeletionCheck{}
+}
+
+type CRDeletionCheck struct{}
```

Comment:
You can consider put `context`, `client` as struct field, so that not just an empty struct, and then `Run` just need 
 to pass one `Object` param

Reply:
The problem here is when we init the options here, we do not have the runtime client nor the context, thus I would leave it as it is - Otherwise we would need to do a bigger refactoring - but since we plan to get rid of declarative in general, this refactoring is obsolite.

Reply:
make sense